### ISSUE TITLE
Can't run unit 3.1 test

### ISSUE THREAD

**Issue by johnthad** [hojberg/mastering-react#23]
When I try to run `npm test` in unit 3.1, it fails with `Error: cannot resolve path (or pattern) '—recursive'`. Using `--recursive` also fails (see below). npm is installing mocha 2.3.3, but I get a similar error if I tweak package.json to the install mocha 2.2.5.

```
thad@localhost:~/archives/web-videos/reactjs-src/Section 3/Video 3.1
615 $ npm install
...

thad@localhost:~/archives/web-videos/reactjs-src/Section 3/Video 3.1
616 $ npm run build

> mastering-react@1.0.0 build /Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1
> browserify src/main.js -t babelify --outfile public/bundle.js && node-sass --output-style compressed styles/main.scss public/main.css

Rendering Complete, saving .css file...
Wrote CSS to /Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1/public/main.css

thad@localhost:~/archives/web-videos/reactjs-src/Section 3/Video 3.1
617 $ npm run test

> mastering-react@1.0.0 test /Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1
> ./node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive


  error: unknown option `--'
```

(A single dash also shows a stack trace.)

The npm-debug.log is 

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/Users/thad/.npm-packages/bin/npm', 'run', 'test' ]
2 info using npm@3.3.5
3 info using node@v0.12.7
4 verbose run-script [ 'pretest', 'test', 'posttest' ]
5 info lifecycle mastering-react@1.0.0~pretest: mastering-react@1.0.0
6 silly lifecycle mastering-react@1.0.0~pretest: no script for pretest, continuing
7 info lifecycle mastering-react@1.0.0~test: mastering-react@1.0.0
8 verbose lifecycle mastering-react@1.0.0~test: unsafe-perm in lifecycle true
9 verbose lifecycle mastering-react@1.0.0~test: PATH: /Users/thad/.npm-packages/lib/node_modules/npm/bin/node-gyp-bin:/Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1/node_modules/.bin:/opt/local/bin:/opt/local/sbin:/Users/thad/.npm-packages/bin:/Users/thad/bin:/Library/Frameworks/Python.framework/Versions/3.4/bin:/opt/groovy/bin:/opt/gradle/bin:/opt/asf/maven/bin:/opt/asf/ant/bin:/Library/Java/JavaVirtualMachines/jdk1.8.0_60.jdk/Contents/Home/bin:/Users/optix/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/Library/Frameworks/Firebird.framework/Resources/bin
10 verbose lifecycle mastering-react@1.0.0~test: CWD: /Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1
11 silly lifecycle mastering-react@1.0.0~test: Args: [ '-c',
11 silly lifecycle   './node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive' ]
12 silly lifecycle mastering-react@1.0.0~test: Returned: code: 1  signal: null
13 info lifecycle mastering-react@1.0.0~test: Failed to exec test script
14 verbose stack Error: mastering-react@1.0.0 test: `./node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive`
14 verbose stack Exit status 1
14 verbose stack     at EventEmitter.<anonymous> (/Users/thad/.npm-packages/lib/node_modules/npm/lib/utils/lifecycle.js:233:16)
14 verbose stack     at EventEmitter.emit (events.js:110:17)
14 verbose stack     at ChildProcess.<anonymous> (/Users/thad/.npm-packages/lib/node_modules/npm/lib/utils/spawn.js:24:14)
14 verbose stack     at ChildProcess.emit (events.js:110:17)
14 verbose stack     at maybeClose (child_process.js:1015:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
15 verbose pkgid mastering-react@1.0.0
16 verbose cwd /Users/thad/archives/web-videos/reactjs-src/Section 3/Video 3.1
17 error Darwin 14.5.0
18 error argv "node" "/Users/thad/.npm-packages/bin/npm" "run" "test"
19 error node v0.12.7
20 error npm  v3.3.5
21 error code ELIFECYCLE
22 error mastering-react@1.0.0 test: `./node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive`
22 error Exit status 1
23 error Failed at the mastering-react@1.0.0 test script './node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive'.
23 error This is most likely a problem with the mastering-react package,
23 error not with npm itself.
23 error Tell the author that this fails on your system:
23 error     ./node_modules/mocha/bin/mocha --compilers js:babel/register --reporter dot -—recursive
23 error You can get their info via:
23 error     npm owner ls mastering-react
23 error There is likely additional logging output above.
24 verbose exit [ 1, true ]
```

**Comment by johnthad** [comment]
Argh! In changing the single dash to double-dash, Atom on the Mac was inserting a em-dash.
Now the only issue is the time assertion, which may be a newer setting in moment (2.10.6 installed vs 2.10.3).

```
      -  "Mar 9, 1:14pm"
      +  "Mar 9, 6:14pm"
```

**Comment by hojberg** [comment]
Oh thats unfortunate.
I believe the time difference is a timezone issue - Surely a brittle test that could have been done better.

**Comment by johnthad** [comment]
Aha! I took a look at the original file. It's not a hyphen (`0x2D`) but `0xE28094` before the word `recursive` (offset `0xE4`). I get that from from the Packt source download or from a copy-paste of package.json from here. Adding a second hyphen makes it look like an em-dash.

Yes, I realized what the timezone issue is. `orderedAt: moment("2015-03-09T17:14:00.000Z").utcOffset(1)` is one way to "fix" it.

Thanks for the video lesson. It's well done. I'm completely new to ReactJS, and I'm finding the lessons interesting.
